data-prepper-script/src/main/antlr/DataPrepperScript.g4

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+grammar DataPrepperScript;
+
+@header {
+    package org.opensearch.dataprepper.script.antlr;
+}
+
+fragment
+Digit
+    : '0'
+    | NonZeroDigit
+    ;
+
+fragment
+NonZeroDigit
+    : [1-9]
+    ;
+
+Integer
+    : '0'
+    | NonZeroDigit Digit*
+    ;
+
+Float
+    : NonZeroDigit? Digit '.' Digit
+    | NonZeroDigit? Digit '.' Digit+ NonZeroDigit
+    | '.' Digit
+    | '.' Digit* NonZeroDigit
+    ;
+
+Boolean
+    : TRUE
+    | FALSE
+    ;
+
+fragment
+StringCharacters
+    : StringCharacter+
+    ;
+
+fragment
+StringCharacter
+    : ~["\\]
+    | EscapeSequence
+    ;
+
+fragment
+EscapeSequence
+    : '\\' [btnfr"'\\]
+    ;
+
+fragment
+JsonPointerEscapeSequence
+    : '\\' [btnfr"'\\/]
+    ;
+
+fragment
+JsonPointerStringCharacters
+    : JsonPointerStringCharacter+
+    ;
+
+fragment
+JsonPointerStringCharacter
+    : ~["\\]
+    | JsonPointerEscapeSequence
+    ;
+
+fragment
+JsonPointerCharacter
+    : [A-Za-z0-9_]
```

Comment:
Here is a valid key that is not supported by this syntax: `a1K.k3-01_02`

Reply:
Okay, Let's confirm and align with our usage of a pointer which do/don't require escaped json pointers to create a consistent user experience across processors.

Reply:
I opened an RFC to update syntax specifics #1005